### PR TITLE
Avoid possible conflict between enum's name and built-in name property

### PR DESCRIPTION
**Pull request by lppedd**
A Kotlin enum has a built-in [name](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-enum/name.html) property, which might conflict with a parser rule named "name".  
~Capitalizing generated enum names for the parser's `Rules` enum solves the issue in an elegant way.~

This fixes this issue and any related issues by replacing the enums with objects containing const values.  
Const values are inlined at compile time, providing a general performance boost while parsing, especially in JS.

cc @ftomassetti 

Closes: #145

### CODE REVIEW

**Comment by lppedd** [comment]
Evaluating a different approach to optimize speed on slower platforms like JS.

**Comment by lppedd** [comment]
@piacenti have a look at this PR. Let me know if it looks good.

You should now be able to also use `var` and `val` as parser rule names.

**Review by piacenti — APPROVED** [review]
I think it looks good

File: antlr-kotlin-target/src/main/kotlin/org/antlr/v4/codegen/target/KotlinTarget.kt [hunk 1]
```diff
@@ -21,20 +22,22 @@ public class KotlinTarget(codeGenerator: CodeGenerator) : JavaTarget(codeGenerat
       "if",
       "implements",
       "import",
-      "is",
       "interface",
+      "internal",
```
Comment (ftomassetti) [inline antlr-kotlin-target/src/main/kotlin/org/antlr/v4/codegen/target/KotlinTarget.kt:26]: This seems to contain a mix of hard keywords, soft keywords, and modifier keywors (see [Kotlin's keyords](https://kotlinlang.org/docs/keyword-reference.html)? )

I would make revise this list based on the documentation. For example, `as` and `break` are missing

**Comment by ftomassetti** [comment]
Looks good to me, I would just revise the list of keywords as per my comment

File: antlr-kotlin-target/src/main/kotlin/org/antlr/v4/codegen/target/KotlinTarget.kt (hunk 1 above)
Comment (ftomassetti) [reply antlr-kotlin-target/src/main/kotlin/org/antlr/v4/codegen/target/KotlinTarget.kt]: Great, thank you!

**Pull request merged**